### PR TITLE
​/container-images 404 error code fix

### PR DESCRIPTION
**Pull request by shreekarSS**
## Related Issues and Dependencies
Fixes: https://github.com/thoth-station/user-api/issues/1780
…

## This Pull Request implements
Implement https://github.com/thoth-station/user-api/issues/1780#issuecomment-1231898531

### CODE REVIEW

File: thoth/user_api/api_v1.py [hunk 1]
```diff
@@ -291,6 +291,21 @@ def list_thoth_container_images(
...
+        return (
+            {"error": "No Image found", "parameters": parameters},
+            404,
+            {
+                "page": page,
+                "per_page": per_page,
+                "page_count": page_count,
+                "entries_count": entries_count,
+                "next": next_page,
+                "prev": prev_page,
```
Comment (KPostOffice) [inline thoth/user_api/api_v1.py:305]: I don't think this dictionary is necessary here.

Comment (shreekarSS) [reply thoth/user_api/api_v1.py]: ack

File: thoth/user_api/api_v1.py [hunk 2]
```diff
@@ -291,20 +291,19 @@ def list_thoth_container_images(
...
         },
         200,
-        {
-            "page": page,
-            "per_page": per_page,
-            "page_count": page_count,
-            "entries_count": entries_count,
-            "next": next_page,
-            "prev": prev_page,
-        },
```
Comment (harshad16) [inline thoth/user_api/api_v1.py:307]: The values are needed here, so user can request the result from next page if needed.

File: thoth/user_api/api_v1.py [hunk 3]
```diff
@@ -291,20 +291,19 @@ def list_thoth_container_images(
...
+            404,
+        )
+
     return (
         {
             "container_images": entries,
             "parameters": parameters,
         },
         200,
-        {
```
Comment (KPostOffice) [inline thoth/user_api/api_v1.py:300]: This is still needed here as navigation info. Return type should be a `Union`

**Review by harshad16 — CHANGES_REQUESTED** [review]
noticed that the header dict is removed ,
it should be present , in 404 contion pass empty dict

File: thoth/user_api/api_v1.py (hunk 2 above)
Comment (shreekarSS) [reply thoth/user_api/api_v1.py]: Yep reverted

File: thoth/user_api/api_v1.py (hunk 3 above)
Comment (shreekarSS) [reply thoth/user_api/api_v1.py]: Fixed it

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve
thanks :100:

**Review by mayaCostantini — APPROVED** [review]
lgtm, thanks :100:

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve
thanks

**Pull request merged**